src/compiler/checker.ts

```diff
@@ -727,6 +727,30 @@ module ts {
             return links.target;
         }
 
+        function resolveAliasAsValue(symbol: Symbol): Symbol {
+            Debug.assert((symbol.flags & SymbolFlags.Alias) !== 0, "Should only get Alias here.");
+            let links = getSymbolLinks(symbol);
+            if (!links.aliasValueTarget) {
+                links.aliasValueTarget = resolvingSymbol;
+                let node = getDeclarationOfAliasSymbol(symbol);
+                let target = resolveAlias(symbol);
+                if ((target.flags & SymbolFlags.Alias) && !(target.flags & SymbolFlags.Value)) {

```

Comment:
can you doc this bit to make it clearer why we're checking this, and why calling resolveAliasAsValue recursively is the right thing to do here.